test/unit/utils/telemetry/collector_test.rb

```diff
@@ -0,0 +1,46 @@
+require 'inspec/utils/telemetry'
+require_relative '../../../helper.rb'
+
+class TestTelemetryCollector < Minitest::Test
+  def setup
+    @collector = Inspec::Telemetry::Collector.instance
+    @collector.reset
+  end
+
+  def test_collector_singleton
+    assert_equal Inspec::Telemetry::Collector.instance, @collector
+  end
+
+  def test_add_data_series
+    data_series = Inspec::Telemetry::DataSeries.new('/resource/File')
+    assert @collector.add_data_series(data_series)
+  end
+
+  def test_list_data_series
+    assert @collector.list_data_series
+  end
+
+  def test_find_or_create_data_series
+    dg = @collector.find_or_create_data_series(:deprecation_group)
+    assert_kind_of Inspec::Telemetry::DataSeries, dg
+    assert_equal :deprecation_group, dg.name
+    assert_equal @collector.find_or_create_data_series(:deprecation_group), dg
+  end
+
+  def test_list_data_series_names
+    data_series = Inspec::Telemetry::DataSeries.new('/resource/File')
+    data_series2 = Inspec::Telemetry::DataSeries.new('/resource/Dir')
+    @collector.add_data_series(data_series)
+    @collector.add_data_series(data_series2)
+    list = @collector.list_data_series_by_name('/resource/File')
```

Comment:
I don't understand why this would take a parameter, or if it does take aparameter, why it would be an Array return. As data series names are unique, if you ask for something by name, you would only ever get back one dataseries.

I could see how having a `list_data_series_names` method (takes no args) could be useful, but I don't see how this is useful by itself.

Reply:
Ok